### PR TITLE
Media: add a Guided Tour for media basics

### PR DESCRIPTION
**Pull request by jerrysarcastic**
This PR adds a tour that shows the basics of the Media Library to people. 

To test:
- start the tour by going to your site's media library and adding `?tour=mediaBasicsTour` to the URL (or just go here and select a site: https://calypso.live/media?branch=add/media-basics-tour&tour=mediaBasicsTour)
- check whether all steps display correctly (cf. GIF in comments below)
- check whether tour is understandable
- check whether steps look nice enough
- check whether the copy could be improved

TODO: 
- [x] guard against sites with no media files
- [x] send out request for another round of reviews

### CODE REVIEW

**Comment by mcsf** [comment]
Add me as a reviewer once you're ready. :)

**Comment by lsinger** [comment]
Here's a GIF of the current state: 

media-basics-tour

**Comment by lancewillett** [comment]
Yay! Nice to see this.

Will Tracks events work "automagically"? To see open and completion events.

**Comment by lsinger** [comment]
> Will Tracks events work "automagically"?

Yes, those are baked into the framework. We just need the tour name (and step names, if we want) to figure those things out.

**Comment by michelleweber** [comment]
Nice! I <3 Guided Tours. Took a spin through, and had a few notes on flow and copy.

------

Opening card suggestion: 
"Welcome to your media library! 

Upload media -- photos, documents, audio files, and more -- by clicking the ( ) add new button."

------

Drop-and-Drop: looks good!

------

Next, I would suggest an interim screen that explains (1) why you're telling me to do these next things and (2) reassure me that I'm not about to make permanent changes to my blog. Suggest:
"Once you upload a file, you can edit its title, add a caption, and even do basic photo editing. To find these options, first click on a file to select it."

"Then, click the Edit button."

(Also, what if there are no images in my library?) 

------

In the editing screen, the first pop-up hides the edit image button for me, which feels odd.

Why does it not dump me back in my library at the end? It feels a little unfinished. Suggest an ending card along the lines of:
"That's it! Click on ( ) media library to head back to your full library. Happy uploading!"

**Comment by lsinger** [comment]
Thanks @michelleweber and @mcsf for your comments! With the last few changes, I tried to address them all, with a few caveats. E.g., I opted to center the last suggested step and not connect it to the "Done" button, since positioning was very unreliable here ("above" often is, unfortunately). 

> Also, what if there are no images in my library?

Luckily users will usually have headstarted images in their libraries, but nonetheless your raise a good point. I think we want to add a selector that will guard us against that — adding a note to the PR summary above.

**Comment by lsinger** [comment]
The guard against empty media libraries should work now, and I've added a few additional copy tweaks. An updated GIF is here:

media-basics-tour-v2

Thankful for a re-review @michelleweber!

**Comment by michelleweber** [comment]
Looks great!

**Comment by jblz** [comment]
My user did not get any images Headstarted, so the tour did not kick off until I added an image then reloaded /media, so that felt a bit strange...

Would it be possible to add a step at the beginning to upload a media item?

**Comment by jblz** [comment]
Is it possible for the tour element to be hidden at the "Edit Image" step if you actually click the `Edit Image` button?

See below, the tour seems out of place when shown above the Image Editor:

screen shot 2017-07-14 at 4 06 08 pm
screen shot 2017-07-14 at 4 06 26 pm

**Comment by lsinger** [comment]
Thanks for the review, @jblz! 

> My user did not get any images Headstarted, so the tour did not kick off until I added an image then reloaded /media, so that felt a bit strange...Would it be possible to add a step at the beginning to upload a media item?

I can see how in a v2 of the tour we could add that step for people who don't have any images. For now though I'd prefer to keep it as it is. Requiring people to upload an image I think is an action where we'd see lots of drop-off. If that assumption holds, we'd negate the potential value of the rest of the tour for many. 

> Is it possible for the tour element to be hidden at the "Edit Image" step if you actually click the Edit Image button?

I looked at hiding the tour step behind the image editor via an inline style for the step, but both the editor modal and the image properties modal have the same z-index. That is, if I hide the tour step behind the image editor, it will be hidden behind the whole image properties modal as well. 
I'd prefer we don't let a tour influence how we build the actual UI, so I'd like to avoid changing a z-index just to make it work with the tour. 

Another option could be having a framework-level mechanism to make the tour quit on purpose here. There's no such mechanism currently. 

In light of the above, I'd be inclined to leave it as it is for this PR. Thoughts?

**Comment by lsinger** [comment]
Went through the tour again with @jblz, together decided to merge the final two steps and reword to avoid any confusion as to whether we want you to click "Edit Image". Updated version: 

media-basics-tour-v3

File: client/layout/guided-tours/tours/media-basics-tour.js [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+		<Step
+			name="drag-and-drop"
+			placement="right"
+		>
+			<p>
+				{ translate( 'You can also drag-and-drop image and video files from your computer into your media library.' ) }
+			</p>
+			<img
+				src="https://en-support.files.wordpress.com/2017/07/media-drag-and-drop.gif"
```
Comment (jblz) [inline client/layout/guided-tours/tours/media-basics-tour.js]: This should be served from the CDN

Comment (mcsf) [inline client/layout/guided-tours/tours/media-basics-tour.js]: (thus: https://i0.wp.com/en-support.files.wordpress.com/2017/07/media-drag-and-drop.gif)

Comment (lsinger) [inline client/layout/guided-tours/tours/media-basics-tour.js]: Done!

Comment (spen) [inline client/layout/guided-tours/tours/media-basics-tour.js]: I assume this should also? I could pop open a quick PR if so:

screen shot 2017-08-08 at 20 42 29

Comment (lsinger) [inline client/layout/guided-tours/tours/media-basics-tour.js]: https://github.com/Automattic/wp-calypso/pull/17014 :)

**Comment by lsinger** [comment]
@jblz tweaked the selector for the image to "the first image that is not yet selected, and if there is none, the first image."

File: client/layout/guided-tours/tours/media-basics-tour.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+		<Step
+			name="drag-and-drop"
+			placement="right"
+		>
+			<p>
+				{ translate( 'You can also drag-and-drop image and video files from your computer into your media library.' ) }
+			</p>
+			<img
+				src="https://i0.wp.com/en-support.files.wordpress.com/2017/07/media-drag-and-drop.gif"
```
Comment (lsinger) [inline client/layout/guided-tours/tours/media-basics-tour.js]: 🤗

**Pull request merged**